src/client/Client.cc

```diff
@@ -8675,10 +8675,6 @@ int Client::_write(Fh *f, int64_t offset, uint64_t size, const char *buf,
   // use/adjust fd pos?
   if (offset < 0) {
     lock_fh_pos(f);
-    /*
-     * FIXME: this is racy in that we may block _after_ this point waiting for caps, and size may
-     * change out from under us.
-     */
```

Comment:
My understanding (and if I'm wrong, please let me know) is that at the end of the _getattr(), we'll hold the shared caps that were requested (Fs in this case). Since we won't release the mutex again before we issue the write, we'll still hold Fs when we go to do the write.

Is that wrong?